src/pint/models/parameter.py

```diff
@@ -1418,6 +1418,24 @@ def __repr__(self):
 
         return out
 
+    def create_new_index_copy(self, index):
+        """Sets up copy of current param with a new index."""
+        new_mask_param = maskParameter(
```

Comment:
@luojing1211 the `new_param()` function does not copy over the `key` or `key_value` parameters. Should I simply add that to `new_param()`? I'm not sure if it would break something somewhere else but I can test it.

Reply:
Let us fix it in an other PR.

Reply:
Actually, why don't we fix it here? The key and key values should only be used in the mask parameter.   It should not break anything else.